.github/workflows/test.yml

```diff
@@ -33,6 +33,8 @@ jobs:
     steps:
       - name: Check out repo
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I believe this will fix workflow to build off of the PR branch. This is what the password manager app has. 

However, it doesn't appear to be pulling in this value. I'm pretty sure it needs to live in main to get picked up properly.